### PR TITLE
Refactor process running code: use RsResult instead of exceptions

### PR DESCRIPTION
**Pull request by vlad20012**
Use `RsResult<ProcessOutput, RsProcessExecutionException>` instead of `@Throws(ExecutionException::class)`

### CODE REVIEW

**Review by mchernyavsky — APPROVED** [review]
LGTM

Since `CleanConfiguration` has a lot in common with `RsResult`, I have refactored the corresponding code too.

File: src/main/kotlin/org/rust/openapiext/CommandLineExt.kt [hunk 1]
```diff
@@ -36,32 +38,29 @@ fun GeneralCommandLine(path: Path, withSudo: Boolean = false, vararg args: Strin
 fun GeneralCommandLine.withWorkDirectory(path: Path?) = withWorkDirectory(path?.systemIndependentPath)
 
 fun GeneralCommandLine.execute(timeoutInMilliseconds: Int?): ProcessOutput? {
```
Comment (avrong) [inline src/main/kotlin/org/rust/openapiext/CommandLineExt.kt:40]: Is there a reason why don't you use `RsProcessResult<ProcessOutput>` here?

Comment (vlad20012) [reply src/main/kotlin/org/rust/openapiext/CommandLineExt.kt:40]: Well, I don't know. I just preserved the signature and semantics of this method.

**Comment by vlad20012** [comment]
bors r=mchernyavsky

**Pull request merged**